### DISCUSSION TITLE
Uploading win10 local source code to a remote server prompts HTTP ERROR 500 【请求帮助！！！谢谢！】

### DISCUSSION THREAD

**Discussion by stuffren** [PrestaShop/PrestaShop#36228] [Q&A]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I have created website in win10 local wsl2 using prestashop open source program, which currently works fine on win10.
I uploaded the source code from the root directory of the website /var/www/ucoredata to the /var/www/ucoredata.tech directory on the remote server via scp.
At the same time, I also created a mysql database prestashop on the remote server and imported the local sql database into prestashop.
I used the virtual domain ucoredata.com locally for testing and development.
I have used the real domain ucoredata.tech on the remote server
I have changed the virtual domain ucoredata.com to www.ucoredata.tech in the remote database.

Currently when I visit www.ucoredata.tech through my browser, it prompts:
This page is not functioning properly www.ucoredata.tech cannot process this request at this time.
HTTP ERROR 500

Looking forward to your help and answers, thanks in advance! Thanks in advance!

Visit the frontend
![01](https://github.com/PrestaShop/PrestaShop/assets/136626456/1133b19c-24fa-4119-a8ef-b8501d94f868)

Visit Backend
![02](https://github.com/PrestaShop/PrestaShop/assets/136626456/34313e11-169c-4647-9198-9a28aa16cb60)

### Expected behavior

The site can be accessed normally

### Steps to reproduce

Environment test: Upload environment test file
scp phppsinfo.php stuffren@192.3.112.214:/var/www/ucoredata.tech/
![03-phppsinfo](https://github.com/PrestaShop/PrestaShop/assets/136626456/185eec15-bc8f-48e5-9b9d-4cc200d61db4)

As you can see based on the screenshot, my configuration almost always matches the environment configuration officially required by prestashop.

Step 1. Package the source code in win10 local wsl2

Step 2: Upload the packaged file to the server using the scp command

Step 3: Unzip the file

Step 4: Create a new database
Name, login and password of the mysql database in the remote server.
Keep the same name, login name and password of the database in the remote server as the mysql database in the local wsl2.

### PrestaShop version(s) where the bug happened

8.1.5;I didn't do the upgrade, it has nothing to do with the upgrade

### PHP version(s) where the bug happened

PHP 8.1.2-1ubuntu2.17 (cli) (built: May  1 2024 10:10:07) (NTS)

### If your bug is related to a module, specify its name and its version

no

### Your company or customer's name goes here (if applicable).

no

**Comment by SharakPL** [comment]
First of all you may want to get a [fresh copy of phppsinfo](https://raw.githubusercontent.com/PrestaShop/php-ps-info/master/phppsinfo.php) from GitHub to have current requirements.

When you upload your store from localhost to a remote server there are few places that need to be updated.
1. In your database dump change any occurrence (there's at least 4) of your localhost shop domain to match the online domain eg. change localhost:8001 to mydomain.com
2. On your server the prime requirement is to update app/config/parameters.php. Even if you use the same database settings as on your localhost, you still need to make sure the host and port are correct. Servers don't always use localhost or 127.0.0.1 with port 3306.
3. If you packed all files to be sent to your server then you need to make sure to remove the old cache. Just clear all the contents of var/cache/

You may also want to update your .htaccess in the root of your shop to also match the new shop domain, but you could also regenerate it from the admin panel.

You also need to regenerate mail templates because they also include references to localhost domain.

**Comment by stuffren** [comment]
> First of all you may want to get a fresh copy of phppsinfo from GitHub to have current requirements.
> 
> When you upload your store from localhost to a remote server there are few places that need to be updated.
> 
> 1. In your database dump change any occurrence of your localhost shop domain to match the online domain eg. change localhost:8001 to mydomain.com
> 2. On your server the prime requirement is to update app/config/parameters.php. Even if you use the same database settings as on your localhost, you still need to make sure the host is correct. Servers don't always use localhost or 127.0.0.1
> 3. If you packed all files to be sent to your server then you need to make sure to remove the old cache. Just clear all the contents of var/cache/
> 
> You may also want to update your .htaccess in the root of your shop to also match the new shop domain, but you could also regenerate it from the admin panel.
> 
> You also need to regenerate mail templates because they also include references to localhost.

----------------------
Thanks @SharakPL, for your patience, I followed the first, second and third steps you told me.
1. in the database, replaced all the fields of ucoredata.com with www.ucoredata.tech
2. in app/config/parameters.php, I made sure that the database address is localhost, because I can access http://Server IP Server IP address/phpmyadmin/ normally, and the phpmyadmin homepage shows the server address as server: localhost:3306. the login name and password I configured are working fine.
3. I deleted all the cache files in the var/cache/ directory

Finally I restarted apache2 and mysql.
Still can't access www.ucoredata.tech

Can you please help me see where else I need to tweak or modify?

If it was you who needed to upload the prestashop source code and mysql database, from your local wsl2 to a remote server (lamp environment), what are the necessary steps you would take?

**Comment by SharakPL** [comment]
Make sure everything is exactly as it should be. Is your domain name `ucoredata.tech` or `www.ucoredata.tech`. You'll get into the redirections loop if you get it wrong.

Try to access your admin panel first.

**Comment by stuffren** [comment]
> Make sure everything is exactly as it should be. Is your domain name `ucoredata.tech` or `www.ucoredata.tech`. You'll get into the redirections loop if you get it wrong.
> 
> Try to access your admin panel first.

Currently accessing the admin panel, it still prompts:
![02](https://github.com/PrestaShop/PrestaShop/assets/136626456/ba17ee06-cd36-4941-a51b-1091bf788377)

**Comment by stuffren** [comment]
This is the configuration file for my website

ucoredata.tech.conf
<VirtualHost *:80>
    ServerAdmin ucoredata@hotmail.com
    ServerName ucoredata.tech
    ServerAlias www.ucoredata.tech
    DocumentRoot /var/www/ucoredata.tech
    ErrorLog ${APACHE_LOG_DIR}/error.log
    CustomLog ${APACHE_LOG_DIR}/access.log combined
RewriteEngine on
RewriteCond %{SERVER_NAME} =ucoredata.tech [OR]
RewriteCond %{SERVER_NAME} =www.ucoredata.tech
RewriteRule ^ https://%{SERVER_NAME}%{REQUEST_URI} [END,NE,R=permanent]
</VirtualHost>

ucoredata.tech-le-ssl.conf
<IfModule mod_ssl.c>
<VirtualHost *:443>
    ServerAdmin ucoredata@hotmail.com
    ServerName ucoredata.tech
    ServerAlias www.ucoredata.tech
    DocumentRoot /var/www/ucoredata.tech
    ErrorLog ${APACHE_LOG_DIR}/error.log
    CustomLog ${APACHE_LOG_DIR}/access.log combined

Include /etc/letsencrypt/options-ssl-apache.conf
SSLCertificateFile /etc/letsencrypt/live/ucoredata.tech/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/ucoredata.tech/privkey.pem
</VirtualHost>
</IfModule>

Can you please help me to see if it will conflict with the prestashop configuration?

**Comment by stuffren** [comment]
Sincerely, I have a question for you:
If it's you, and you need to upload the prestashop source code and sql files that already have data from win10 local wsl2 to ubuntu server (lamp environment), what are the steps that you would take in order to get the site up and running?

**Comment by SharakPL** [comment]
@stuffren I already explained required steps in my [1st post](https://github.com/PrestaShop/PrestaShop/issues/36201#issuecomment-2121748924) 😉

If it doesn't work for you then either you have misconfiguration in your **app/config/parameters.php** or within your server config.
First of all I recommend setting `memory_limit` to at least 512M.
Then enable dev mode to get more specifics on the error instead of just error code 500. Make sure error_reporting is enabled on your server and then change **config/defines.inc.php** from
![obraz](https://github.com/PrestaShop/PrestaShop/assets/5007456/e3f99211-4044-4704-9300-3eafa14d0777)
to
![obraz](https://github.com/PrestaShop/PrestaShop/assets/5007456/c6276945-6640-4ebf-86f4-ed6339bb1378)

**Reply by SharakPL** [reply]
@stuffren did you try this? What's the actual error you get?

**Comment by stuffren** [comment]
Even though my problem was not solved, I still appreciate your response! Thank you!

Finally, I suggest prestashop official to write a tutorial on **《How to publish win10 local prestashop source code and mysql database to server》**.

**Reply by SharakPL** [reply]
Good idea, though it's not specific to win10 or any windows installation. It's just about moving the shop from one server to another. Each server has its own specifications so the shop needs to be adjusted for new configuration, new domain, etc.

**Comment by SharakPL** [comment]
@stuffren also a very important security requirement ‼️

| Since you publicly shared your admin url, better make sure to change the name of the admin folder when you're done fixing your configuration.<br>You should NEVER share your admin URL or e-mail or password publicly! 
--|

**Comment by stuffren** [comment]
@SharakPL 
Thank you for your continued tracking of my issue and patience in helping me find possible solutions!

I haven't tried setting memory_limit to at least 512M, and I haven't enabled error_reporting either.I've tried all the solutions you've mentioned before to no avail, and I understand that you guys are usually busy with other things as well, and can't possibly just fix my problem, but I don't want to wait around any longer.

Therefore, the solution I took was to reinstall it directly on the server side, and after a successful installation, the site is now working fine! This suggests that the problem I'm experiencing is not due to memory limitations. But it's clear that this is not a good approach, and it could even be argued that this is a bad approach, if you look at it in the long run.
Try to think about it, if I already have 10,000 pieces of data on my site, but can't migrate the site's data to a local or other server, that's terrible.

This time fortunately there is less data locally. The re-installation is only suitable for cases where the local data is small.

Whether you are migrating from local (source code and SQL data) to remote server, or from remote server (source code and SQL data) to local, it would be a tragedy for your company's website operation and management if you have a large amount of data and there are major errors in the migration that can't be solved in a short period of time.

Therefore, I **strongly recommend** prestashop to write a tutorial on **《How to publish local source code and mysql data to remote server》**.

And I thought of another tip: migrating local prestashop source code and data to a remote server is essentially a data recovery thing, which should be as simple as recovering our daily address book. The day before yesterday, I looked at the apache error log, constantly prompting “prestashop PHP source code” related to the error. This shows that the architectural design of prestashop may not have taken into account the need for users to “migrate” source code and data.

After all this, I really hope that prestashop can become better, lighter and simpler!
Finally, thanks for reminding me about the security risk of admin url. I will change it immediately!

**Reply by SharakPL** [reply]
Fresh PS 8 should work perfectly fine with 128M memory limit. 512M is my personal recommendation based on experience with a few modules that tend to throw memory allocation error because they do sometimes need more than officially recommended 256M. Your error 500 is most certainly not related to memory_limit.
Like I mentioned already it's most likely a misconfiguration on your part. So once again please check
- if you're green on **phppsinfo**
- if your database credentials are correct for your remote server in **app/config/parameters.php**
- if your sql dump has domain names updated and is imported to new database
- if **var/cache/** folder is empty before the first run

If your shop doesn't work after that, then you've made a mistake somewhere.
I made hundreds of shop transfers between localhost > test server > destination server and can assure you the above steps are all you need to make your shop work. Then just update .htaccess to load images properly and regenerate mail themes.

If you don't have the patience to check these steps or enable error reporting then I can't help you.

**Reply by stuffren** [reply]
Thank you @SharakPL for your patient reply!
It's great to see your heartily edited reply based on your own experience.
Especially when you mentioned, “You did hundreds of store transfers between localhost > test server > target server, as well as making sure that all the steps you used were all that was needed to make the store work properly”. I can define it as the official prestashop tutorial: **《How to publish local source code and mysql data to a remote server》**. Haha!

I'll definitely find some time to test it again by following exactly the steps you mentioned!

Did you know? @SharakPL, I stand as a prestashop user and love your kind of reply based on your own experience. Because such replies are more likely to reassure users that their steps have been verified many times over. There are guarantees!

**Comment by ChillCode** [comment]


**Reply by stuffren** [reply]
Thank you @ChillCode for your reply!
The “key folder unwritable” issue you mentioned was actually discovered by myself during my interaction with SharakPL. Once I realized that my site root directory was not writable, I gave the server user write permissions and then followed the steps mentioned by @SharakPL again, but the site was still inaccessible.
Only I didn't communicate with him on github about the issue.

Regarding your mention of creating the site directly on the server instance, this is fine though. But my habit is to debug the site locally first (e.g. change some theme details, modify some page templates) and then upload it to the remote server.
Also usually we need to migrate the source code and data from the remote server to local.

So it's a matter of migrating between local and remote servers, not a single issue of building the site on the server or locally.

Finally, about the official prestashop [《How to migrate to PrestaShop 8》](https://devdocs.prestashop-project.org/8/basics/keeping-up-to-date/migration/#via-the-import-feature-of-prestashop) that you mentioned, I've read it as well. This tutorial is written in a more complex way and it is not as simple as the migration steps that @SharakPL told me.
